myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -96,6 +101,48 @@ public Response flexUp(FlexUpClusterRequest request) {
 
     @Timed
     @PUT
+    @Path("/flexupservice")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response flexUpservice(FlexUpServiceRequest request) {
+      Preconditions.checkNotNull(request,
+              "request object cannot be null or empty");
+
+      LOGGER.info("Received Flexup a Service Request");
+
+      Integer instances = request.getInstances();
+      String profile = request.getServiceName();
+
+      LOGGER.info("Instances: {}", instances);
+      LOGGER.info("Profile: {}", profile);
+      
+      // Validate profile request
+      Response.ResponseBuilder response = Response.status(Response.Status.ACCEPTED);
+      
+      if (cfg.getServiceConfiguration(profile) == null) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Service does not exist: " + profile);
+        LOGGER.error("Provided service does not exist " + profile);
+        return response.build();
+      }
+      
+      if (!this.isValidInstanceSize(instances)) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Invalid instance size: " + instances);
+        LOGGER.error("Invalid instance size request " + instances);
+        return response.build();
+      }
+
+      try {
+        this.myriadOperations.flexUpAService(instances, profile);
+      } catch (MyriadBadConfigurationException e) {
```

Comment:
Started making changes and realized we still need it - for example we started cluster with maxInstances for a service = 2, then we flexed up 2 more they will start since validation only checks for instances > 0. This is not the most significant one. Most interesting is when RM fails over and tries to start again 2 instances that are already running - which should be prevented. It is easier to make checks in a single place like MyriadOperations and not in two places: ClustersResource and Main